source/simulator/assemblers/newton_stokes.cc

```diff
@@ -588,7 +588,7 @@ namespace aspect
               for (unsigned int i=0; i<stokes_dofs_per_cell; ++i)
                 for (unsigned int j=0; j<stokes_dofs_per_cell; ++j)
                   {
-                    data.local_matrix(i,j) += (- eta_two_thirds * (scratch.div_phi_u[i] * scratch.div_phi_u[j])) * JxW;
+                    data.local_matrix(i,j) -= (- eta_two_thirds * (scratch.div_phi_u[i] * scratch.div_phi_u[j])) * JxW;
```

Comment:
@MFraters I find this change confusing. The old version is identical to the term in the normal Stokes equation (e.g. stokes.cc:532). Does this term have a different sign for the Newton matrix?

I wonder if the convergence issues are instead related to #1967? I just checked and it seems there is no compressibility correction (i.e. the term in this code) for the Newton Stokes preconditioner, which could negatively affect convergence behavior.